recipes_source/distributed_async_checkpoint_recipe.rst

```diff
@@ -0,0 +1,259 @@
+Asynchronous Saving with Distributed Checkpoint (DCP)
+=====================================================
+
+Checkpointing is often a bottle-neck in the critical path for distributed training workloads, incurring larger and larger costs as both model and world sizes grow.
+One excellent strategy for offsetting this cost is to checkpoint in parallel, asynchronously. Below, we expand the save example
+from the `Getting Started with Distributed Checkpoint Tutorial <https://github.com/pytorch/tutorials/blob/main/recipes_source/distributed_checkpoint_recipe.rst>`__
+to show how this can be integrated quite easily with `torch.distributed.checkpoint.async_save`.
+
```

Comment:
Can you add a what you will learn and Prerequistes sections? You could use [cards](https://github.com/pytorch/tutorials/blob/main/beginner_source/template_tutorial.py#L9-L23) for this. 
For example:

What you will learn
* How to use parallel checkpoints asynchronously
* Effective strategies to optimize performance

Prerequisites
* PyTorch 2.4 or later
* Any other prereqs like understanding of async_save or other